test/strategies/explicit.jl

```diff
@@ -0,0 +1,54 @@
+good = KNNClassifier(K=2)
+bad = KNNClassifier(K=10)
+ugly = ConstantClassifier()
+
+r = [good, bad, ugly]
+
+rng=StableRNGs.StableRNG(123)
+X, y = make_blobs(rng=rng)
+
+@testset "constructor" begin
+true 
+end
+
+
+@testset_accelerated "integration" accel begin
+
+    # evaluate the three models separately:
+    resampling = Holdout(fraction_train=0.6)
+    scores = map(r) do model
+        e = evaluate(model, X, y, resampling=resampling, measure=LogLoss())
+        return e.measurement[1]
+    end
+    # scores are ordered best to worst:
+    # 3-element Vector{Float64}:
+    #  2.2204460492503136e-16
+    #  0.005578588782855459
+    #  1.121552757665965
+
+    # find the scores using `Explicit` tuning:
+    tmodel = TunedModel(model=first(r),
```

Comment:
I don't understand why `first(r)` is passed to the model and all the models `r` are passed to `range`.

Reply:
In the new syntax I could drop the `model=first(r)` but under the hood this assignment is indeed needed. A `TunedModel` has a type-parameter, which is the type of the atomic model being cloned and mutated in all the other strategies. The reason we need this is so that the wrapped `TunedModel` type can inherit all the traits (eg, `target_scitype`) of the atomic model. In the special `Explicit` case there is no single model type in general, so we just pick the first one, unless the user specifies it explicitly, as here (because I forgot to change this to the newer syntax, which is now corrected). 

This is not perfect, because there is nothing preventing the user from having models whose types have different traits, which could lead to unexpected behaviour. An alternative would be to force the type parameter to be the abstract supertype `Probabilistic` or `Deterministic` in the special case of the `Explicit` strategy, in which case the various traits fall back to things like `Unknown`. The reason I did not do this was because there might be cases where the iterator of models *are* all of the same type, and it would be useful to preserve faithful trait information. We could accomodate both the homogeneous and inhomogeneous type cases, but only by adding logic to iterate over all models to check which is the case, and I did not want to restrict to finite iterators. 

On reflection, it's probably better to live with the abstract type parameter for all `Explicit` cases for now - choosing `Deterministic` or `Probabilistic` depending on the type of the first model (the implicit assumption that they all share the same one of these as a supertype, which is already a documented assumption).